docs/doxygen/porting.txt

```diff
@@ -0,0 +1,77 @@
+/**
+@page mqtt_porting Porting Guide
+@brief Guide for porting MQTT to a new platform.
+
+A port to a new system must provide the following components:
+1. [Configuration Macros](@ref porting_config)
+2. [Transport Interface](@ref porting_transport)
+3. [Time Function](@ref porting_time)
+
+@section porting_config Configuration Macros
+@brief Settings that must be set as macros in the config header `core_mqtt_config.h`, or passed in as compiler options
+
+@see [Configurations](@ref core_mqtt_config)
+
+@note Regardless of whether the following macros are defined in `core_mqtt_config.h` or passed as compiler options,
+the `core_mqtt_config.h` file must be included to build the MQTT library.
+
+The following macros can be configured for the managed MQTT library:
+ - @ref MQTT_STATE_ARRAY_MAX_COUNT <br>
+ - @ref MQTT_PINGRESP_TIMEOUT_MS <br>
+ - @ref MQTT_MAX_CONNACK_RECEIVE_RETRY_COUNT
```

Comment:
There can be a separate line to refer to the `MQTT_DO_NOT_USE_CUSTOM_CONFIG` doxygen tag that is populated in the core_mqtt_config page [here](https://github.com/FreeRTOS/coreMQTT/blob/master/docs/doxygen/pages.txt#L65)

Reply:
Added